src/include/sof/lib/clk.h

```diff
@@ -21,6 +21,7 @@ struct timer;
 #define CLOCK_NOTIFY_POST	1
 
 struct clock_notify_data {
+	uint32_t clk_id;
```

Comment:
This is getting more complex here, could you add some inline comments for each field here.